test_scripts/Resumption/Handling_errors_from_HMI/027_resume_all_data_with_one_error_rpc_unexpected_disconnect_different_resultCodes.lua

```diff
@@ -0,0 +1,92 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0190-resumption-data-error-handling.md
+--
+-- Requirement summary:TBD
+--
+-- Description:
+-- 1. AddSubMenu for resumption is added by app
+-- 2. Unexpected disconnect and reconnect are performed
+-- 3. App reregisters with actual HashId
+-- 4. UI.AddSubMenu request is sent from SDL to HMI during resumption
+-- 5. HMI responds with error_n resultCode to UI.AddSubMenu request
+-- SDL does:
+-- 1. process unsuccess response from HMI
+-- 2. respond RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED) to mobile application
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Resumption/Handling_errors_from_HMI/commonResumptionErrorHandling')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local resultCodes = {
+  "UNSUPPORTED_REQUEST",
+  "UNSUPPORTED_RESOURCE",
+  "DISALLOWED",
+  "REJECTED",
+  "ABORTED",
+  "IGNORED",
+  "RETRY",
+  "IN_USE",
+  "DATA_NOT_AVAILABLE",
+  "TIMED_OUT",
+  "INVALID_DATA",
+  "CHAR_LIMIT_EXCEEDED",
+  "INVALID_ID",
+  "DUPLICATE_NAME",
+  "APPLICATION_NOT_REGISTERED",
+  "WRONG_LANGUAGE",
+  "OUT_OF_MEMORY",
+  "TOO_MANY_PENDING_REQUESTS",
+  "NO_APPS_REGISTERED",
+  "NO_DEVICES_CONNECTED",
+  "USER_DISALLOWED",
+  "TRUNCATED_DATA",
+  "READ_ONLY"
+}
+
+--[[ Local Functions ]]
+local function reRegisterApp(pAppId, pErrorCode)
+  local mobSession = common.getMobileSession(pAppId)
+  mobSession:StartService(7)
+  :Do(function()
+      local params = common.cloneTable(config["application" .. pAppId].registerAppInterfaceParams)
+      params.hashID = common.hashId[pAppId]
+      local corId = mobSession:SendRPC("RegisterAppInterface", params)
+      common.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered", {
+          application = { appName = config["application" .. pAppId].registerAppInterfaceParams.appName }
+        })
```

Comment:
@GetmanetsIrina It's possible to use `common.getConfigAppParams()` instead of `config["application" .. pAppId].registerAppInterfaceParams`